### PR TITLE
feat: [OeX_Proctoring-178] (Proctored exam epic) add mock api for fetching proctoring settings

### PR DESCRIPTION
**Pull request by ihor-romaniuk**
**Description:** add mock API for fetching proctoring settings and proctoring entrance instruction

**Youtrack:** https://youtrack.raccoongang.com/issue/OeX_Proctoring-178

### CODE REVIEW

File: src/exam/Sequence.jsx [hunk 1]
```diff
@@ -37,8 +40,12 @@ SequenceExamWrapper.propTypes = {
...
   loadExamData: PropTypes.func.isRequired,
+  loadProctoringSetting: PropTypes.func.isRequired,
   courseId: PropTypes.string.isRequired,
   children: PropTypes.element.isRequired,
 };
 
-export default withExamStore(SequenceExamWrapper, null, { loadExamData: getExamAttemptsData });
+export default withExamStore(SequenceExamWrapper, null, {
+  loadExamData: getExamAttemptsData,
+  loadProctoringSetting: getProctoringSettings,
```
Comment (TamoshaytisV) [inline src/exam/Sequence.jsx]:
```suggestion
  loadProctoringSettings: getProctoringSettings,
```

File: src/exam/Sequence.jsx [hunk 2]
```diff
@@ -37,8 +40,12 @@ SequenceExamWrapper.propTypes = {
     isTimeLimited: PropTypes.bool,
   }).isRequired,
   loadExamData: PropTypes.func.isRequired,
+  loadProctoringSetting: PropTypes.func.isRequired,
```
Comment (TamoshaytisV) [inline src/exam/Sequence.jsx]:
```suggestion
  loadProctoringSettings: PropTypes.func.isRequired,
```

File: src/exam/Sequence.jsx [hunk 3]
```diff
@@ -17,11 +17,14 @@ import { getExamAttemptsData } from '../data';
...
 const SequenceExamWrapper = ({ children, ...props }) => {
   const {
-    sequence, courseId, loadExamData, ...examProps
+    sequence, courseId, loadExamData, loadProctoringSetting, ...examProps
   } = props;
 
   useEffect(() => {
-    loadExamData(courseId, sequence.id);
+    (async () => {
+      await loadProctoringSetting();
```
Comment (TamoshaytisV) [inline src/exam/Sequence.jsx]:
```suggestion
      await loadProctoringSettings();
```

File: src/exam/Sequence.jsx [hunk 4]
```diff
@@ -17,11 +17,14 @@ import { getExamAttemptsData } from '../data';
  */
 const SequenceExamWrapper = ({ children, ...props }) => {
   const {
-    sequence, courseId, loadExamData, ...examProps
+    sequence, courseId, loadExamData, loadProctoringSetting, ...examProps
```
Comment (TamoshaytisV) [inline src/exam/Sequence.jsx]:
```suggestion
    sequence, courseId, loadExamData, loadProctoringSettings, ...examProps
```

File: src/exam/Sequence.jsx (hunk 3 above)
Comment (TamoshaytisV) [inline src/exam/Sequence.jsx]: Why do we await in this line but don't in the next one? Can we call both asyncroniously?

File: src/instructions/index.jsx [hunk 5]
```diff
@@ -3,26 +3,41 @@ import { ExamStatus } from '../constants';
...
-    case attempt.attempt_status === ExamStatus.READY_TO_SUBMIT:
-      return <SubmitExamInstructions />;
-    case attempt.attempt_status === ExamStatus.SUBMITTED:
-      return <SubmittedExamInstructions />;
-    default:
-      return children;
+const Instructions = ({
+  attempt, examHasAttempt, isProctored, children,
+}) => {
+  if (isProctored) {
```
Comment (TamoshaytisV) [inline src/instructions/index.jsx]: Can we use the common logic like this?
```js
switch (true) {
  case !examHasAttempt:
    return isProctored
      ? <EntranceProctoredExamInstructions />
      : <StartExamInstructions />;
  case attempt.attempt_status === ExamStatus.CREATED:
    return <VerificationProctoredExamInstructions />;
  case attempt.attempt_status === ExamStatus.READY_TO_SUBMIT:
    return <SubmitExamInstructions />;
  case attempt.attempt_status === ExamStatus.SUBMITTED:
    return <SubmittedExamInstructions />;
  default:
    return children;
```

File: src/instructions/proctored_exam/EntranceProctoredExamInstructions.jsx [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+            id="exam.startExamInstructions.footerButton"
+            defaultMessage="Can I request additional time to complete my exam?"
+          />
+        </Button>
+      </p>
+    </div>
+  </div>
+);
+
+const mapExamStateToProps = (state) => {
```
Comment (TamoshaytisV) [inline src/instructions/proctored_exam/EntranceProctoredExamInstructions.jsx]: Do we need `mapExamStateToProps` at all? I see you are not using this values in props

File: src/instructions/proctored_exam/VerificationProctoredExamInstructions.jsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    <div className="footer-sequence">
+      <p>
+        <Button
+          data-testid="request-exam-time-button"
+          variant="link"
+          onClick={() => {}}
+        >
+          <FormattedMessage
+            id="exam.startExamInstructions.footerButton"
+            defaultMessage="Can I request additional time to complete my exam?"
```
Comment (TamoshaytisV) [inline src/instructions/proctored_exam/VerificationProctoredExamInstructions.jsx]: Is this message really present on verification instructions page?

File: src/instructions/proctored_exam/VerificationProctoredExamInstructions.jsx [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+          />
+        </Button>
+      </p>
+    </div>
+  </div>
+);
+
+export default withExamStore(
```
Comment (TamoshaytisV) [inline src/instructions/proctored_exam/VerificationProctoredExamInstructions.jsx]: Why do we need to wrap the component with `withExamStore`? You do not use any store state values

**Review by TamoshaytisV — CHANGES_REQUESTED** [review]

File: src/exam/Sequence.jsx (hunk 3 above)
Comment (ihor-romaniuk) [reply src/exam/Sequence.jsx]: If we don't use async/await construction in the first render the component shows "timed exam instruction" and after that switch on the correct "proctoring instruction". Maybe you have a more correct way to solve it.

File: src/exam/Sequence.jsx (hunk 4 above)
Comment (ihor-romaniuk) [reply src/exam/Sequence.jsx]: changed

File: src/exam/Sequence.jsx (hunk 3 above)
Comment (ihor-romaniuk) [reply src/exam/Sequence.jsx]: changed

File: src/exam/Sequence.jsx (hunk 2 above)
Comment (ihor-romaniuk) [reply src/exam/Sequence.jsx]: changed

File: src/exam/Sequence.jsx (hunk 1 above)
Comment (ihor-romaniuk) [reply src/exam/Sequence.jsx]: changed

File: src/instructions/index.jsx (hunk 5 above)
Comment (ihor-romaniuk) [reply src/instructions/index.jsx]: completely agree with you!

File: src/instructions/proctored_exam/EntranceProctoredExamInstructions.jsx (hunk 6 above)
Comment (ihor-romaniuk) [reply src/instructions/proctored_exam/EntranceProctoredExamInstructions.jsx]: Removed. Thanks

File: src/instructions/proctored_exam/VerificationProctoredExamInstructions.jsx (hunk 7 above)
Comment (ihor-romaniuk) [reply src/instructions/proctored_exam/VerificationProctoredExamInstructions.jsx]: I got that instruction text without no connected proctoring providers. Maybe it's a blank instruction and we will see another one with an enabled proctoring provider.

File: src/instructions/proctored_exam/VerificationProctoredExamInstructions.jsx (hunk 8 above)
Comment (ihor-romaniuk) [reply src/instructions/proctored_exam/VerificationProctoredExamInstructions.jsx]: fixed

File: src/exam/Sequence.jsx (hunk 3 above)
Comment (ihor-romaniuk) [reply src/exam/Sequence.jsx]: The structure was changed

**Comment by TamoshaytisV** [comment]
Introduce ENV variable to enable/disable proctored exam support. Goal: disable proctored exams on production unless feature is ready; enable development process locally

File: src/exam/Sequence.jsx [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+ * It takes control over rendering exam instructions unless exam is started only if
+ * current sequence item is timed exam. Otherwise, renders any children elements passed.
+ * @param children - Current course sequence item content (e.g. unit, navigation buttons etc.)
+ * @returns {JSX.Element}
+ * @example
+ * <SequenceExamWrapper sequence={sequence} courseId={courseId}>
+ *   {sequenceContent}
+ * </SequenceExamWrapper>
+ */
+const SequenceExamWrapper = ({ children, ...props }) => {
```
Comment (UvgenGen) [inline src/exam/Sequence.jsx:18]: after refactoring, "SequenceExamWrapper" from "src/exc/Sequence.jsx" is not used. moved to "src/index.jsx"

**Comment by ihor-romaniuk** [comment]
This PR was reopened -> https://github.com/raccoongang/frontend-lib-special-exams/pull/14

**Pull request closed without merging**